core/globals.h

```diff
@@ -1132,8 +1136,12 @@ int
 tolower(int c);
 int
 strcasecmp(const char *left, const char *right);
+int
+strncasecmp(const char *left, const char *right, size_t n);
 unsigned long
 strtoul(const char *str, char **end, int base);
+int
+fnmatch(const char *pattern, const char *string, int flags);
```

Comment:
It would be good to add some unit tests of the new utilities.  See how the existing ones are done: mostly small functions embedded in some files: e.g., see unit_test_io() in core/io.c inside STANDALONE_UNIT_TEST.